clientv3/watch.go

```diff
@@ -484,9 +490,35 @@ func (w *watchGrpcStream) run() {
 				closeErr = err
 				return
 			}
-			if wc, closeErr = w.newWatchClient(); closeErr != nil {
+
+			clientOpenC := make(chan openWatchClientResponse, 1)
+			go func() {
+				owc, err := w.openWatchClient(w.ctx)
+				clientOpenC <- openWatchClientResponse{wc: owc, err: err}
+			}()
+
+			for {
+				select {
+				case opc := <-clientOpenC:
+					wc = opc.wc
+					closeErr = opc.err
+				case ws := <-w.closingc:
+					w.closeSubstream(ws)
```

Comment:
It could make the code a bit complicated, because `closing` is a local variable tied to this loop and in the the `if` block, the is a `return` to break the loop.
If moving the `closing`  variable to the struct makes sense, I'll do that.